src/XenCert/StorageHandler.py

```diff
@@ -54,6 +54,18 @@ def report(predicate, condition):
 # Hardcoded time limit for Functional tests in hours
 timeLimitFunctional = 4
 
+def retval_judge(retval, exception, checkpoint=None, point=None):
+    if not retval:
+        raise Exception(exception)
+    if checkpoint:
+        checkpoint += point
+    return checkpoint
+
+def result_judge(result, exception, uuid=None):
```

Comment:
Should not 'get' uuid firstly then check result, because the 'get' maybe depends on just good result.
You can change (for both retval_judge) like blow:
```
 def check_result(retval, exception):
    if not retval:
        raise Exception(exception)

 def check_result_for_checkpoint(retval, exception, checkpoint=0, point=1):
    check_result(retval, exception)
    return checkpoint + point

```
then all places using result_judge, change to 
```
check_result(retval, exception)
uuid = ... // remain original 
```